### ISSUE TITLE
Consider using fmtlib over printf

### ISSUE THREAD

**Issue by rajsite** [ni/VireoSDK#702]
Has some nice perf wins and can maybe coalesce the via Print %z specifier into it with the custom types support: https://github.com/fmtlib/fmt

**Comment by sanmut** [comment]
I agree this is a good one to use. If we upgrade to compilers supporting C++20, this is part of the standard library, and we would not need external dependency.

**Comment by rajsite** [comment]
Ah cool, didn't know that! Just for my curiosity, here is a SO on some of the libfmt features not in C++ standard library: https://stackoverflow.com/a/63587221/338923
